src/epics/chatEpics.js

```diff
@@ -0,0 +1,210 @@
+import {
+    CHAT_INIT_APPLICATION,
+    CHAT_SET_USERNAME,
+    CHAT_INIT_CONNECTION,
+    CHAT_INIT_CHAT,
+    CHAT_SEND_MESSAGE,
+    readyToConnect,
+    pincodeRequired,
+    useParanoidMode,
+    initConnection,
+    connectionInitiated,
+    disconnected,
+    addMessage,
+    messageSent,
+    setUserStatus,
+    userStatusSent,
+    usernameSet
+} from '../actions/chatActions';
+
+import {
+    alertSuccess,
+    alertWarning
+} from '../actions/alertActions';
+
+import {
+    USER_STATUS_DELAY_MS
+} from '../constants/messaging';
+
+import { getEmail, getPassphrase, generateMessageKey } from '../utils/encrypter';
+import ChatHandler from './helpers/ChatHandler';
+import miniLock from '../utils/miniLock';
+import { combineEpics } from 'redux-observable';
+import createDetectVisibilityObservable from './detectPageVisibilityObservable';
+
+const BACKEND_URL = window.location.origin;
+const authUrl = `${window.location.origin}/api/login`;
+const wsUrl = `${window.location.origin.replace('http', 'ws')}/api/ws/messages/all`;
+const chatHandler = new ChatHandler(wsUrl);
+
+import { Observable } from 'rxjs/Observable';
+import { ajax } from 'rxjs/observable/dom/ajax';
+import 'rxjs/add/operator/partition';
+import 'rxjs/add/operator/mergeMap';
+import 'rxjs/add/operator/delay';
+import 'rxjs/add/operator/throttleTime';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/mapTo';
+import 'rxjs/add/operator/catch';
+import 'rxjs/add/operator/do';
+import 'rxjs/add/operator/retry';
+import 'rxjs/add/operator/pluck';
+import 'rxjs/add/operator/filter';
+import 'rxjs/add/operator/retryWhen';
+import 'rxjs/add/observable/of';
+import 'rxjs/add/observable/from';
+import 'rxjs/add/observable/fromEvent';
+import 'rxjs/add/observable/merge';
+import 'rxjs/add/observable/if';
+
+function createKeyPairObservable({ pincode = '' }) {
+    /* Using a function */
+    const keyPairObservable = Observable.create(function(observer) {
```

Comment:
what is the difference between declaring an observable then returning it, versus just directly returning like you did on line 87

Reply:
There is not much more difference than returning any other anonymous object vs. creating and assigning it first.

This is just me being inconsistent 😅